hubmap_api_py_client/internal.py

```diff
@@ -61,14 +62,17 @@ def hubmap_query(
         '''
         This function takes query parameters and returns a query set token.
         '''
-        self._check_parameters(
-            input_type=input_type, output_type=output_type,
-            input_set=input_set, genomic_modality=genomic_modality, p_value=p_value)
-        request_url = self.base_url + output_type + "/"
-        request_dict = self._fill_request_dict(
-            input_type, input_set, genomic_modality, p_value, logical_operator)
-
-        response = requests.post(request_url, request_dict)
+        if input_type is None:
+            response = requests.get(self.base_url + output_type + "/")
+        else:
+            self._check_parameters(
+                input_type=input_type, output_type=output_type,
+                input_set=input_set, genomic_modality=genomic_modality, p_value=p_value)
+            request_url = self.base_url + output_type + "/"
+            request_dict = self._fill_request_dict(
+                input_type, input_set, genomic_modality, p_value, logical_operator)
+
+            response = requests.post(request_url, request_dict)
```

Comment:
Will the API choke if we send it empty parameters? I really want to avoid adding complexity here. Perhaps move the none check inside _check_parameters?
```suggestion
        self._check_parameters(
            input_type=input_type, output_type=output_type,
            input_set=input_set, genomic_modality=genomic_modality, p_value=p_value)
        request_url = self.base_url + output_type + "/"
        request_dict = self._fill_request_dict(
            input_type, input_set, genomic_modality, p_value, logical_operator)

        response = requests.post(request_url, request_dict)
```
or
```suggestion
        if input_type is not None:
            self._check_parameters(
                input_type=input_type, output_type=output_type,
                input_set=input_set, genomic_modality=genomic_modality, p_value=p_value)
        request_url = self.base_url + output_type + "/"
        request_dict = self._fill_request_dict(
            input_type, input_set, genomic_modality, p_value, logical_operator)

        response = requests.post(request_url, request_dict)
```
or if that's not feasible:
```suggestion
        request_url = self.base_url + output_type + "/"
        if input_type is None:
            response = requests.get(request_url)
        else:
            self._check_parameters(
                input_type=input_type, output_type=output_type,
                input_set=input_set, genomic_modality=genomic_modality, p_value=p_value)
            request_dict = self._fill_request_dict(
                input_type, input_set, genomic_modality, p_value, logical_operator)
            response = requests.post(request_url, request_dict)
```